### PR TITLE
Submodule Statistical Fix

### PR DESCRIPTION
**Pull request by johanjino**
This fix should pass all tests for `jax`, `numpy` and `tensorflow`.

A significant change is added to the `tensorflow` backend:

- The `var()`  and `std()` functions in the backend has been changed to run on a self defined function. This was done to replicate `tensorflow` behavior along with using parameters like correction which or else is not supported by `tensorflow`. 

These changes, if approved can also be implemented on `PyTorch` backend, which will complete and pass all tests

### CODE REVIEW

File: ivy/functional/backends/numpy/__init__.py [hunk 1]
```diff
@@ -89,6 +89,7 @@
 supports_gradients = False
 
 
+
```
Comment (djl11) [inline ivy/functional/backends/numpy/__init__.py:92]: ?

File: ivy/functional/backends/mxnet/__init__.py [hunk 2]
```diff
@@ -60,6 +60,7 @@
 supports_gradients = True
 
 
+
```
Comment (djl11) [inline ivy/functional/backends/mxnet/__init__.py:63]: ?

File: ivy/functional/backends/tensorflow/statistical.py [hunk 3]
```diff
@@ -6,6 +6,18 @@
 # local
 import ivy
 
+def __new_var_fun(x, *, axis, correction, dtype):
```
Comment (djl11) [inline ivy/functional/backends/tensorflow/statistical.py:9]: we instead use a single underscore for private functions, because the name mangling is quite annoying when debugging, and we don't really have a need for "protected" functions

File: ivy/functional/backends/tensorflow/statistical.py [hunk 4]
```diff
@@ -6,6 +6,18 @@
...
+    output = tf.cast(x, dtype)
+    length = tf.cast(tf.shape(output)[axis], dtype)
+    divisor = tf.cast(length - correction, dtype)
+    mean = tf.keras.backend.sum(output, axis=axis) / length
+    output = tf.math.abs(tf.cast(output, dtype=dtype) - tf.cast(tf.expand_dims(mean, axis), dtype=dtype))
+    output = output ** 2
+    output = tf.keras.backend.sum(output, axis=axis) / divisor
+    return output
+
+def __new_std_fun(x, *, axis, correction, dtype):
```
Comment (djl11) [inline ivy/functional/backends/tensorflow/statistical.py:19]: likewise

File: ivy/functional/backends/torch/statistical.py [hunk 5]
```diff
@@ -106,9 +110,9 @@ def prod(
                     ]
                 ),
                 dtype=dtype,
-                out=out,
+                **out
             )
-    return torch.prod(input=x, dim=axis, dtype=dtype, keepdim=keepdims, out=out)
+    return torch.prod(input=x, dim=axis, dtype=dtype, keepdim=keepdims, **out)
```
Comment (djl11) [inline ivy/functional/backends/torch/statistical.py:115]: does passing `out=None` fail here, otherwise, what is the point of this `out` dict?

File: ivy/functional/ivy/gradients.py [hunk 6]
```diff
@@ -356,6 +356,7 @@ def execute_with_gradients(func, xs, retain_grads=False):
     """
     return current_backend(None).execute_with_gradients(func, xs, retain_grads)
 
+execute_with_gradients.computes_gradients = True
```
Comment (djl11) [inline ivy/functional/ivy/gradients.py:359]: ?

File: ivy/functional/ivy/meta.py [hunk 7]
```diff
@@ -489,6 +489,7 @@ def fomaml_step(
         return cost, grads, rets[2]
     return cost, grads
 
+fomaml_step.computes_gradients = True
```
Comment (djl11) [inline ivy/functional/ivy/meta.py:492]: ?

File: ivy/functional/ivy/meta.py [hunk 8]
```diff
@@ -575,6 +576,8 @@ def reptile_step(
         return cost, grads, rets[2]
     return cost, grads
 
+reptile_step.computes_gradients = True
```
Comment (djl11) [inline ivy/functional/ivy/meta.py:579]: ?

**Review by djl11 — CHANGES_REQUESTED** [review]

File: ivy/functional/backends/tensorflow/statistical.py [hunk 9]
```diff
@@ -7,6 +7,24 @@
 import ivy
 
 
+def _new_var_fun(x, *, axis, correction, dtype):
+    output = tf.cast(x, dtype)
+    length = tf.cast(tf.shape(output)[axis], dtype)
+    divisor = tf.cast(length - correction, dtype)
+    mean = tf.keras.backend.sum(output, axis=axis) / length
```
Comment (djl11) [inline ivy/functional/backends/tensorflow/statistical.py:14]: I'm sure tensorflow has a `sum` function without us needing to go via `keras`

**Review by djl11 — CHANGES_REQUESTED** [review]

**Review by djl11 — APPROVED** [review]
looks good to me, but I imagine it might be possible to implement things in a slightly cleaner manner in future. Either way, provided that the tests are all passing, then feel free to merge!

**Comment by johanjino** [comment]
@djl11 

Thanks Dan! Ill sure keep working to make it cleaner before I mark the submodule complete.
